src/index.ts

```diff
@@ -161,6 +161,27 @@ export function isOfShape<V extends Dict, T extends Shape<V> = Shape<V>> (shape:
   return fn as GuardWithShape<Unshape<T>>
 }
 
+/**
+ * Create a validator that asserts that passed argument is a tuple of certain elements.
+ */
+export function isTuple<A>(a: Guard<A>): Guard<[A]>
+export function isTuple<A, B>(a: Guard<A>, b: Guard<B>): Guard<[A, B]>
+export function isTuple<A, B, C>(a: Guard<A>, b: Guard<B>, c: Guard<C>): Guard<[A, B, C]>
+export function isTuple<A, B, C>(a: Guard<A>, b?: Guard<B>, c?: Guard<C>): Guard<[A]> | Guard<[A, B]> | Guard<[A, B, C]>
```

Comment:
What's this additional one for? I don't think we need it. Does it compile without?

Reply:
True. Removed it.